### ISSUE TITLE
camp.route is not documented

### ISSUE THREAD

**Issue by techtonik** [espadrine/sc#50]
I fail to find documentation for `camp.route` params needed to understand `shields.io` code. There is a reference to some "Diving In" section at https://github.com/espadrine/sc/blob/master/doc/Readme.md#ask-and-augmented-request but search doesn't show anything,

**Comment by jankeromnes** [comment]
Hi @techtonik, I'm not sure either what this "Diving in" reference is for (reminds of the unrelated [Walkthrough](https://github.com/espadrine/sc/blob/master/lib/Walkthrough)), but for what it's worth I know and use `camp.route`.

Note: It's progressively being deprecated in favour of `camp.path` (and `Ask` is also being deprecated), which might be why the documentation is gone.

As mentioned in the docs, the parameters are `camp.route(pattern, function (query, path, end, ask) {})` (I prefer to call them `app.route(regex, (data, match, end, query)`, which might or might not be more intuitive to you):

- `pattern`: a regex like `/^\/about\/?$/`
- `query`: data extracted from the incoming request (URL query parameters like `?page=4`, urlencoded POST parameters, or multipart data from `<form>` submits which can include uploading files)
- `path`: actually the matched results from the regex, e.g. `[ '/about' ]`, or `[ '/ABC', 'B', 'C' ]` if your regex is `/\/A\(\w\)\(\w\)`
- `end`: the callback you can use to serve a response. Its parameters are `end(templateData, { template: '/path/to/template-file.html' })`
- `ask`: actually just a container for `req` and `res`. In some cases you might prefer to just call `ask.res.statusCode = 404; ask.res.end('not found')` instead of using the `end()` callback

**Comment by techtonik** [comment]
In shields code it seems that `query` comes decoded - instead of `?page=4` it can be accessed as `query.page`? Is that right? In that case it will be better to rename it to  `params` meaning "URL params extracted from query", because `data` is too generic to have any context meaning.

**Comment by jankeromnes** [comment]
That's true, everything in `query` is decoded (URL query parameters are parsed, POST body parameters are urldecoded, and multipart form data from `<form>` submits are re-assembled).
